onnx/defs/generator/defs.cc

```diff
@@ -51,6 +98,60 @@ ONNX_OPERATOR_SET_SCHEMA(
             return;
           }
 
+          if (nullptr != value_int) {
+            // OpSchema::Verify check ensures that the attribute value has_i():
+            if (!value_int->has_i())
+              fail_shape_inference("Attribute 'value_int' expect an integer.")
+            updateOutputElemType(ctx, 0, TensorProto::INT64);
+            updateOutputShape(ctx, 0, TensorShapeProto());
```

Comment:
An alternative is simply `getOutputShape(ctx, 0)` since the shape attribute doesn't exists yet (I tried to pick the one that was semantically more sounding).

Reply:
Change to "updateOutputShape(ctx, 0, {})". I think "TensorShapeProto()" would represent an unknown shape (of unknown rank). It is more precise to specify output tensor is of rank 0.

Reply:
I think some constructors / types specifications are missing to allow this syntax to compile. :/

Reply:
May be the compiler is unable to infer the type of the parameter from an empty list {}. In any case, I realize that the existing code is fine. I was confusing this with a type without a defined shape ... a TensorShapeProto() should represent a shape with an empty list of dims, so that's fine. Please ignore my previous suggestion.